### ISSUE TITLE
[Android] Stack overflow crash when using dotnet-trace for profiled aot

### ISSUE THREAD

**Issue by steveisok** [dotnet/runtime#71143]
### Description

A stack overflow crash occurs on Android when stack walking in order to gather the right data for MethodDetails events.  It appears to be `class_loading` profiler events which get fired over and over until we overflow the stack and crash.

### Reproduction Steps

1. Start dotnet-dsrouter

Once a device is connected via USB or an emulator is started:

`dotnet-dsrouter server-server -ipcs ~/mylocalport -tcps 127.0.0.1:9000 --forward-port Android`

2. Build runtime and run the Android [sample](https://github.com/dotnet/runtime/tree/main/src/mono/sample/Android)

In the Makefile, uncomment [RUNTIME_COMPONENTS](https://github.com/dotnet/runtime/blob/d3f7e0145437ca3afe7d8a0bab38cc95040cfc28/src/mono/sample/Android/Makefile#L13) and [DIAGNOSTIC_PORTS](https://github.com/dotnet/runtime/blob/d3f7e0145437ca3afe7d8a0bab38cc95040cfc28/src/mono/sample/Android/Makefile#L15)

`make run AOT=true MONO_CONFIG=Debug MONO_ARCH=<target-arch>`

3. Start dotnet-trace

`dotnet-trace collect --diagnostic-port ~/mylocalport,connect --providers Microsoft-Windows-DotNETRuntime:0x1F000080018:5`

### Expected behavior

The trace should run and produce a nettrace file

### Actual behavior

The app crashes. You'll notice the stack trace repeats over and over like:

```
mono_walk_stack_with_ctx mini-exceptions.c:1217
mono_runtime_walk_stack_with_ctx mini-exceptions.c:1194
ep_rt_mono_walk_managed_stack_for_thread ep-rt-mono.c:2650
ep_rt_walk_managed_stack_for_thread ep-rt-mono.h:696
ep_walk_managed_stack_for_current_thread ep.h:236
ep_buffer_manager_write_event ep-buffer-manager.c:974
ep_session_write_event ep-session.c:482
write_event_2 ep.c:768
write_event ep.c:702
ep_write_event ep.c:1494
EventPipeWriteEventTypeLoadStart dotnetruntime.c:802
FireEtwTypeLoadStart clretwallmain.h:1313
ep_rt_mono_write_event_type_load_start ep-rt-mono.c:3998
runtime_profiler_class_loading ep-rt-mono.c:4489
mono_profiler_raise_class_loading profiler-events.h:54
mono_class_create_from_typedef class-init.c:482
mono_class_get_checked class.c:2886
mono_get_method_from_token loader.c:1062
mono_get_method_checked loader.c:1195
mono_aot_find_jit_info aot-runtime.c:3619
mono_jit_info_table_find_internal jit-info.c:330
mini_jit_info_table_find_ext mini-exceptions.c:1583
mono_find_jit_info_ext mini-exceptions.c:668
unwinder_unwind_frame mini-exceptions.c:796
mono_walk_stack_full mini-exceptions.c:1349
mono_walk_stack_with_ctx mini-exceptions.c:1217
mono_runtime_walk_stack_with_ctx mini-exceptions.c:1194
ep_rt_mono_walk_managed_stack_for_thread ep-rt-mono.c:2650
ep_rt_walk_managed_stack_for_thread ep-rt-mono.h:696
ep_walk_managed_stack_for_current_thread ep.h:236
ep_buffer_manager_write_event ep-buffer-manager.c:974
ep_session_write_event ep-session.c:482
write_event_2 ep.c:768
write_event ep.c:702
ep_write_event ep.c:1494
EventPipeWriteEventTypeLoadStart dotnetruntime.c:802
FireEtwTypeLoadStart clretwallmain.h:1313
```

Full stack trace: [message.txt](https://github.com/dotnet/runtime/files/8952494/message.txt)

### Regression?

No

### Known Workarounds

None

### Configuration

Latest main using Android with any architecture we support (arm64, x64, x86)

**Comment by lateralusX** [comment]
Also note that this case is not what profiled AOT will use on Android, since it will run in JIT mode or potential partial AOT and not full AOT. JIT mode is not hit by this issue, since all calls will go over the JIT trampoline, making sure all methods on callstack will end up in JIT info table.

**Comment by lateralusX** [comment]
Should be fixed by #71610.
